solutions/gold/ac-smallmultiple.mdx

```diff
@@ -7,6 +7,39 @@ author: Chongtian Ma
 
 [Official Analysis](https://img.atcoder.jp/arc084/editorial.pdf)
 
+## Aproach
+
+All positive integers can be created by starting from 1 and repeating the
+following two operations.
+
+- Add 1 to the current number. At this time, the sum of each digit increases
+  by 1. (This does not apply when the ones place is 9, but we will show later
+  that you do not need to take this into account when solving this problem.)
+- Multiply the current number by 10. At this time, the sum of each digit does
+  not change.
+
+Now, let's consider a graph with all positive integers as vertices and an edge
+from the "current number" to the "new number" of the above operation. What we
+need to find is the shortest path length on this graph from 1 to any multiple of
+K plus 1.
+
+As it is, the number of vertices is infinite, but if you consider how the edges
+are drawn, you can see that the vertices corresponding to each positive integer
+can be equated with modK.
+
+Therefore, we only need to find the shortest path from 1 to 0 on a graph of K
+vertices, where each vertex is equated with modK, and this can be found in O(K)
+time using 01BFS.
+
+A transition that adds 1 to an integer with a 1's digit of 9 does not need to be
+treated specially because if you recall the shortest path algorithm, you can see
+that the destination vertex has already been visited.
```

Comment:
elaborate on this, how does this relate to the problem?

Reply:
Done